File: src/net/simensan/KahluaTestExample/resources/ExampleTest.lua
local square, size

function before()
    square = CreateSquare(10)
end


function after()
   assertEquals(size, 10)
end


function testCreateSquare()
    assertNotNull(square)
    size = square:GetSize()
end


function testCreateSquare2()
    local square2 = CreateSquare(12)
    size = square:GetSize()
end


function testLuaOnlyTest()
   assertEquals(size, 10)
end


function testLuaOnlyTest2()
   assertNotNull(square)
end

File: src/no/simensan/KahluaTestExample/ExampleTest.java
package no.simensan.KahluaTestExample;

import no.simensan.kahluatest.AbstractKahluaTest;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static se.mockachino.Mockachino.*;

public class ExampleTest extends AbstractKahluaTest {
    private SquareBindings squareBindings;

    @Before
    public void setup() {
        squareBindings = spy(new SquareBindings());
        kahluaVm.getLuaJavaClassExposer().exposeClass(Square.class);
        kahluaVm.getLuaJavaClassExposer().exposeGlobalFunctions(squareBindings);
    }

    @Test
    public void testCreateSquare() {
        callLuaTest();
        verifyOnce().on(squareBindings).createSquare(10);
   }

    @Test
    public void testCreateSquare2() {
        assertEquals(1,1);
        callLuaTest();
        verifyOnce().on(squareBindings).createSquare(12);
   }
}